# Analysis Demo Services Churning of Telco-customer

Project objective and aim
- To identify the key services used by the unique demographic
- To identify the churn rate of each services
- Create a visual representation of the result using Seaborn

## Information on Dataset

Using Telco Customer churn data obtain from Kaggle: https://www.kaggle.com/datasets/blastchar/telco-customer-churn?resource=download.
> The data was downloaded onto a local machine on the  27/01/2025

Features of the dataset 
- Customer ID: Unique code to each customer
- Gender: Whether the customer is a male or a female
- Senior Citizen: Whether the customer is a senior citizen or not (1, 0)
- Partner: Whether the customer has a partner or not (Yes, No)
- Dependents: Whether the customer has dependents or not (Yes, No)
- Tenure: Number of months the customer has stayed with the company
- PoneService: Whether the customer has a phone service or not (Yes, No)
- MultipleLines: Whether the customer has multiple lines or not (Yes, No, No phone service)
- InternetService: Customer’s internet service provider (DSL, Fiber optic, No)
- OnlineSecurity: Whether the customer has online security or not (Yes, No, No internet service)
- OnlineBackup: Whether the customer has online backup or not (Yes, No, No internet service)
- DeviceProtection: Whether the customer has device protection or not (Yes, No, No internet service)
- TechSupport: Whether the customer has tech support or not (Yes, No, No internet service)
- StreamingTV: Whether the customer has streaming TV or not (Yes, No, No internet service)
- StreamingMovies: Whether the customer has streaming movies or not (Yes, No, No internet service)
- Contract: The contract term of the customer (Month-to-month, One year, Two year)
- PaperlessBilling: Whether the customer has paperless billing or not (Yes, No)
- PaymentMethod: The customer’s payment method (Electronic check, Mailed check, Bank transfer (automatic), Credit card
- MonthlyCharges: The amount charged to the customer monthly
- TotalCharges: The total amount charged to the customer
- Churn: Whether the customer churned or not (Yes or No)

## Loading dataset

Import the required libraries.

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

Loading the data

In [3]:
df = pd.read_csv('Telco-Customer-Churn-Data.csv')

In [4]:
Demographic_data = {
    'Demographic Segment': [
        'Segment 1', 'Segment 2', 'Segment 3', 'Segment 4', 'Segment 5', 'Segment 6',
        'Segment 7', 'Segment 8', 'Segment 9', 'Segment 10', 'Segment 11', 'Segment 12',
        'Segment 13', 'Segment 14', 'Segment 15', 'Segment 16'
    ],
    'Description': [
        'Male, Non-Senior Citizen, Has Partner, Has Dependents',
        'Female, Non-Senior Citizen, Has Partner, Has Dependents',
        'Female, Non-Senior Citizen, Has Partner, No Dependents',
        'Male, Senior Citizen, No Partner, Has Dependents',
        'Male, Non-Senior Citizen, No Partner, Has Dependents',
        'Female, Senior Citizen, Has Partner, Has Dependents',
        'Male, Non-Senior Citizen, Has Partner, No Dependents',
        'Female, Non-Senior Citizen, No Partner, Has Dependents',
        'Male, Senior Citizen, Has Partner, Has Dependents',
        'Male, Non-Senior Citizen, No Partner, No Dependents',
        'Female, Non-Senior Citizen, No Partner, No Dependents',
        'Female, Senior Citizen, No Partner, Has Dependents',
        'Female, Senior Citizen, Has Partner, No Dependents',
        'Male, Senior Citizen, Has Partner, No Dependents',
        'Male, Senior Citizen, No Partner, No Dependents',
        'Female, Senior Citizen, No Partner, No Dependents'
    ]
}

Demographic_df = pd.DataFrame(Demographic_data)
Demographic_df.to_csv('Telco-Customer-Demographic-data.csv', index = False)
Demographic_df

Demographic Segment                                        Description
0            Segment 1  Male, Non-Senior Citizen, Has Partner, Has Dep...
1            Segment 2  Female, Non-Senior Citizen, Has Partner, Has D...
2            Segment 3  Female, Non-Senior Citizen, Has Partner, No De...
3            Segment 4   Male, Senior Citizen, No Partner, Has Dependents
4            Segment 5  Male, Non-Senior Citizen, No Partner, Has Depe...
5            Segment 6  Female, Senior Citizen, Has Partner, Has Depen...
6            Segment 7  Male, Non-Senior Citizen, Has Partner, No Depe...
7            Segment 8  Female, Non-Senior Citizen, No Partner, Has De...
8            Segment 9  Male, Senior Citizen, Has Partner, Has Dependents
9           Segment 10  Male, Non-Senior Citizen, No Partner, No Depen...
10          Segment 11  Female, Non-Senior Citizen, No Partner, No Dep...
11          Segment 12  Female, Senior Citizen, No Partner, Has Depend...
12          Segment 13  Female, Senior Citizen, Has Partner, No Depend...
13          Segment 14   Male, Senior Citizen, Has Partner, No Dependents
14          Segment 15    Male, Senior Citizen, No Partner, No Dependents
15          Segment 16  Female, Senior Citizen, No Partner, No Dependents

## Data Exploring the data set

Displaying the top 5 rows and columns of the dataset to get a feel of how the dataset look.

In [5]:
df.head()

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   
4          No          No              No  Month-to-month              Yes   

               PaymentMethod MonthlyCharges  TotalCharges Churn  
0           Electronic check          29.85         29.85    No  
1               Mailed check          56.95        1889.5    No  
2               Mailed check          53.85        108.15   Yes  
3  Bank transfer (automatic)          42.30       1840.75    No  
4           Electronic check          70.70        151.65   Yes  

[5 rows x 21 columns]

In [6]:
# Exploring the numbers of rows and columns of data in the dataset.
df.shape

(7043, 21)

### Getting more information on the data

Exploring the dataset and identify the data types of each column.

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 



Note:
* The data set have 7043 observation and 21 variable.
* There is no missing data
* TotalCharges is an object datatype, from observation, this should be numerical data type (float or int) 

In [8]:
# changing the data type of TotalCharges from object to float and printing data types of all the columns to confirm change.

df["TotalCharges"] = pd.to_numeric(df["TotalCharges"], errors='coerce')
df["TotalCharges"] = df["TotalCharges"].astype("float")

# print the list of column and their data type to confirmed that totalcharges has been changed to a numerical datatype.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 


## Data Wrangling

### Searching duplicates Values

In [9]:
df.duplicated().value_counts()

False    7043
Name: count, dtype: int64

No duplicate find

### Searching Missing Values

In [10]:
missing_data = df.isnull().sum()
missing_data

customerID           0
gender               0
SeniorCitizen        0
Partner              0
Dependents           0
tenure               0
PhoneService         0
MultipleLines        0
InternetService      0
OnlineSecurity       0
OnlineBackup         0
DeviceProtection     0
TechSupport          0
StreamingTV          0
StreamingMovies      0
Contract             0
PaperlessBilling     0
PaymentMethod        0
MonthlyCharges       0
TotalCharges        11
Churn                0
dtype: int64

Based on the above result, it can be observed that TotalCharges, have 11 missing data.
> This has no baring on feature of interesting, and in this instance it will be largely ignore. But explored to find the reason of the missing data.

#### Finding line with the missing values

In [11]:
null_mask = df.isnull().any(axis=1)
null_rows = df[null_mask]

null_rows

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
488   4472-LVYGI  Female              0     Yes        Yes       0   
753   3115-CZMZD    Male              0      No        Yes       0   
936   5709-LVOEQ  Female              0     Yes        Yes       0   
1082  4367-NUYAO    Male              0     Yes        Yes       0   
1340  1371-DWPAZ  Female              0     Yes        Yes       0   
3331  7644-OMVMY    Male              0     Yes        Yes       0   
3826  3213-VVOLG    Male              0     Yes        Yes       0   
4380  2520-SGTTA  Female              0     Yes        Yes       0   
5218  2923-ARZLG    Male              0     Yes        Yes       0   
6670  4075-WKNIU  Female              0     Yes        Yes       0   
6754  2775-SEFEE    Male              0      No        Yes       0   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
488            No  No phone service             DSL                  Yes  ...   
753           Yes                No              No  No internet service  ...   
936           Yes                No             DSL                  Yes  ...   
1082          Yes               Yes              No  No internet service  ...   
1340           No  No phone service             DSL                  Yes  ...   
3331          Yes                No              No  No internet service  ...   
3826          Yes               Yes              No  No internet service  ...   
4380          Yes                No              No  No internet service  ...   
5218          Yes                No              No  No internet service  ...   
6670          Yes               Yes             DSL                   No  ...   
6754          Yes               Yes             DSL                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
488                   Yes                  Yes                  Yes   
753   No internet service  No internet service  No internet service   
936                   Yes                   No                  Yes   
1082  No internet service  No internet service  No internet service   
1340                  Yes                  Yes                  Yes   
3331  No internet service  No internet service  No internet service   
3826  No internet service  No internet service  No internet service   
4380  No internet service  No internet service  No internet service   
5218  No internet service  No internet service  No internet service   
6670                  Yes                  Yes                  Yes   
6754                   No                  Yes                   No   

          StreamingMovies  Contract PaperlessBilling  \
488                    No  Two year              Yes   
753   No internet service  Two year               No   
936                   Yes  Two year               No   
1082  No internet service  Two year               No   
1340                   No  Two year               No   
3331  No internet service  Two year               No   
3826  No internet service  Two year               No   
4380  No internet service  Two year               No   
5218  No internet service  One year              Yes   
6670                   No  Two year               No   
6754                   No  Two year              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges  Churn  
488   Bank transfer (automatic)          52.55           NaN     No  
753                Mailed check          20.25           NaN     No  
936                Mailed check          80.85           NaN     No  
1082               Mailed check          25.75           NaN     No  
1340    Credit card (automatic)          56.05           NaN     No  
3331               Mailed check          19.85           NaN     No  
3826               Mailed check          25.35           NaN     No  
4380               Mailed check          20.00           NaN     No  
5218               Mailed check     

#### Replacing Missing Values

In [12]:
df.replace({'TotalCharges':np.nan}, 0, inplace = True)

missing_data = df.isnull().sum()
missing_data

customerID          0
gender              0
SeniorCitizen       0
Partner             0
Dependents          0
tenure              0
PhoneService        0
MultipleLines       0
InternetService     0
OnlineSecurity      0
OnlineBackup        0
DeviceProtection    0
TechSupport         0
StreamingTV         0
StreamingMovies     0
Contract            0
PaperlessBilling    0
PaymentMethod       0
MonthlyCharges      0
TotalCharges        0
Churn               0
dtype: int64

From the observation of the table above, the conclusion is the missing data are for new customer at the beginning of their contract, as a result have zero tenure, and consequently have no total charge record.

## Analysis

### Creating a Filtering variable for each unique demographic 

In [13]:
Unique_demo = {
    'Demographic Segment': [
        'Segment 1', 'Segment 2', 'Segment 3', 'Segment 4', 'Segment 5', 'Segment 6',
        'Segment 7', 'Segment 8', 'Segment 9', 'Segment 10', 'Segment 11', 'Segment 12',
        'Segment 13', 'Segment 14', 'Segment 15', 'Segment 16'
    ],
    'Filter' : [(df['gender']=='Male')&(df['SeniorCitizen']==0)&(df['Partner']=='Yes')&(df['Dependents']=='Yes'),
                (df['gender']=='Female')&(df['SeniorCitizen']==0)&(df['Partner']=='Yes')&(df['Dependents']=='Yes'),
                (df['gender']=='Female')&(df['SeniorCitizen']==0)&(df['Partner']=='Yes')&(df['Dependents']=='No'),
                (df['gender']=='Male')&(df['SeniorCitizen']==1)&(df['Partner']=='No')&(df['Dependents']=='Yes'),
                (df['gender']=='Male')&(df['SeniorCitizen']==0)&(df['Partner']=='No')&(df['Dependents']=='Yes'),
                (df['gender']=='Female')&(df['SeniorCitizen']==1)&(df['Partner']=='Yes')&(df['Dependents']=='Yes'),
                (df['gender']=='Male')&(df['SeniorCitizen']==0)&(df['Partner']=='Yes')&(df['Dependents']=='No'),
                (df['gender']=='Female')&(df['SeniorCitizen']==0)&(df['Partner']=='No')&(df['Dependents']=='Yes'),
                (df['gender']=='Male')&(df['SeniorCitizen']==1)&(df['Partner']=='Yes')&(df['Dependents']=='Yes'),
                (df['gender']=='Male')&(df['SeniorCitizen']==0)&(df['Partner']=='No')&(df['Dependents']=='No'),
                (df['gender']=='Female')&(df['SeniorCitizen']==0)&(df['Partner']=='No')&(df['Dependents']=='No'),
                (df['gender']=='Female')&(df['SeniorCitizen']==1)&(df['Partner']=='No')&(df['Dependents']=='Yes'),
                (df['gender']=='Female')&(df['SeniorCitizen']==1)&(df['Partner']=='Yes')&(df['Dependents']=='No'),
                (df['gender']=='Male')&(df['SeniorCitizen']==1)&(df['Partner']=='Yes')&(df['Dependents']=='No'),
                (df['gender']=='Male')&(df['SeniorCitizen']==1)&(df['Partner']=='No')&(df['Dependents']=='No'),
                (df['gender']=='Female')&(df['SeniorCitizen']==1)&(df['Partner']=='No')&(df['Dependents']=='No')]
              }

unique_demo_df = pd.DataFrame(Unique_demo)
unique_demo_df

Demographic Segment                                             Filter
0            Segment 1  0       False
1       False
2       False
3   ...
1            Segment 2  0       False
1       False
2       False
3   ...
2            Segment 3  0        True
1       False
2       False
3   ...
3            Segment 4  0       False
1       False
2       False
3   ...
4            Segment 5  0       False
1       False
2       False
3   ...
5            Segment 6  0       False
1       False
2       False
3   ...
6            Segment 7  0       False
1       False
2       False
3   ...
7            Segment 8  0       False
1       False
2       False
3   ...
8            Segment 9  0       False
1       False
2       False
3   ...
9           Segment 10  0       False
1        True
2        True
3   ...
10          Segment 11  0       False
1       False
2       False
3   ...
11          Segment 12  0       False
1       False
2       False
3   ...
12          Segment 13  0       False
1       False
2       False
3   ...
13          Segment 14  0       False
1       False
2       False
3   ...
14          Segment 15  0       False
1       False
2       False
3   ...
15          Segment 16  0       False
1       False
2       False
3   ...

### Financial Analysis for each unique demographic

In [28]:
Summary = {
    'Demographic Segment': [],
    'Customer Count': [],
    'Average Tenure (Months)': [],
    'Average Monthly Charge': [],
    'Monthly Revenue Before Churn':[],
    'Churned Customer Count' : [],
    'Churn Rate (%)': [],
    'Monthly Revenue Lost': [],
    'Monthly Revenue After Churn' : [],
    'Percentage difference (%)' : []
}



for i in range(len(unique_demo_df)):
    group_name = unique_demo_df.loc[i, 'Demographic Segment']
    filter_condition = unique_demo_df.loc[i, 'Filter']

    Monthly_revenue = round(df[filter_condition]['MonthlyCharges'].sum(), 2)
    Summary['Demographic Segment'].append(group_name)
    Summary['Customer Count'].append(df[filter_condition].shape[0])
    Summary['Average Tenure (Months)'].append(round(df[filter_condition]['tenure'].mean(), 2))
    Summary['Average Monthly Charge'].append(round(df[filter_condition]['MonthlyCharges'].mean(), 2))
    Summary['Monthly Revenue Before Churn'].append(Monthly_revenue)

    dummy = df[filter_condition]
    churn_rate = round(((dummy[dummy['Churn'] == 'Yes'].shape[0]) / (dummy.shape[0])) * 100) if dummy.shape[0] > 0 else 0
    Summary['Churned Customer Count'].append(dummy[dummy['Churn'] == 'Yes'].shape[0])
    Summary['Churn Rate (%)'].append(churn_rate)

    churned_customers = dummy[dummy['Churn'] == 'Yes']
    total_lost_revenue = churned_customers['MonthlyCharges'].sum() if not churned_customers.empty else 0
    Monthly_revenue_churned = round((df[filter_condition]['MonthlyCharges'].sum())- (total_lost_revenue), 2)
    p_d_monthly_rev_mnthly_churn = round(((Monthly_revenue - Monthly_revenue_churned) /((Monthly_revenue + Monthly_revenue_churned)/2))*100, 2)
    Summary['Monthly Revenue Lost'].append(round(total_lost_revenue, 2))
    Summary['Monthly Revenue After Churn'].append(Monthly_revenue_churned)
    Summary['Percentage difference (%)'].append(p_d_monthly_rev_mnthly_churn)
    

# Create the Pandas DataFrame
Summary_df = pd.DataFrame(Summary)

# Sort the DataFrame by 'Total Churn Revenue Lost' in descending order
Summary_df_sorted = Summary_df.sort_values(by=['Monthly Revenue Lost'], ascending=False)

Summary_df.to_csv('Telco-Customer-Analysis-Summar.csv', index = False)

Summary_df_sorted

Demographic Segment  Customer Count  Average Tenure (Months)  \
10          Segment 11            1338                    22.96   
9           Segment 10            1381                    22.89   
15          Segment 16             317                    24.89   
6            Segment 7             567                    43.98   
14          Segment 15             244                    25.43   
2            Segment 3             596                    43.32   
1            Segment 2             844                    40.60   
13          Segment 14             281                    39.15   
0            Segment 1             822                    41.57   
12          Segment 13             209                    42.33   
4            Segment 5             211                    25.14   
7            Segment 8             142                    22.04   
8            Segment 9              44                    49.45   
5            Segment 6              39                    44.26   
11          Segment 12               3                    21.33   
3            Segment 4               5                    22.60   

    Average Monthly Charge  Monthly Revenue Before Churn  \
10                   61.30                      82016.35   
9                    58.98                      81453.15   
15                   78.36                      24840.80   
6                    73.53                      41691.75   
14                   74.90                      18276.35   
2                    69.96                      41697.30   
1                    59.02                      49809.80   
13                   80.96                      22751.15   
0                    60.76                      49941.65   
12                   85.16                      17798.00   
4                    49.83                      10513.15   
7                    55.20                       7838.60   
8                    84.80                       3731.30   
5                    80.87                       3153.90   
11                   92.55                        277.65   
3                    65.14                        325.70   

    Churned Customer Count  Churn Rate (%)  Monthly Revenue Lost  \
10                     429              32              30805.85   
9                      418              30              28354.30   
15                     158              50              12324.10   
6                      128              23              10566.70   
14                     118              48               9221.40   
2                      114              19               9062.45   
1                      124              15               9015.40   
13                     105              37               8876.25   
0                      105              13               7992.20   
12                      73              35               6165.45   
4                       43              20               2749.65   
7                       32              23               2164.70   
8                       12              27               1071.65   
5                        8              21                639.60   
11                       1              33                 71.00   
3                        1              20                 50.15   

    Monthly Revenue After Churn  Percentage difference (%)  
10                     51210.50                      46.25  
9                      53098.85                      42.15  
15                     12516.70                      65.98  
6                      31125.05                      29.02  
14                      9054.95                      67.48  
2                      32634.85                      24.38  
1                      40794.40                      19.90  
13                     13874.90                      48.47  
0                      41949.45                      17.39  
12                     11632.55                      41.90  
4             

In [29]:
Total_lost = round(Summary_df_sorted['Monthly Revenue Lost'].sum(),2)
Total_lost

139130.85

#### Financial Analysis Sorted By Monthly Revenue before Churn

In [30]:
Summary_df_sorted = Summary_df.sort_values(by=['Monthly Revenue Before Churn'], ascending=False)
Summary_df_sorted

Demographic Segment  Customer Count  Average Tenure (Months)  \
10          Segment 11            1338                    22.96   
9           Segment 10            1381                    22.89   
0            Segment 1             822                    41.57   
1            Segment 2             844                    40.60   
2            Segment 3             596                    43.32   
6            Segment 7             567                    43.98   
15          Segment 16             317                    24.89   
13          Segment 14             281                    39.15   
14          Segment 15             244                    25.43   
12          Segment 13             209                    42.33   
4            Segment 5             211                    25.14   
7            Segment 8             142                    22.04   
8            Segment 9              44                    49.45   
5            Segment 6              39                    44.26   
3            Segment 4               5                    22.60   
11          Segment 12               3                    21.33   

    Average Monthly Charge  Monthly Revenue Before Churn  \
10                   61.30                      82016.35   
9                    58.98                      81453.15   
0                    60.76                      49941.65   
1                    59.02                      49809.80   
2                    69.96                      41697.30   
6                    73.53                      41691.75   
15                   78.36                      24840.80   
13                   80.96                      22751.15   
14                   74.90                      18276.35   
12                   85.16                      17798.00   
4                    49.83                      10513.15   
7                    55.20                       7838.60   
8                    84.80                       3731.30   
5                    80.87                       3153.90   
3                    65.14                        325.70   
11                   92.55                        277.65   

    Churned Customer Count  Churn Rate (%)  Monthly Revenue Lost  \
10                     429              32              30805.85   
9                      418              30              28354.30   
0                      105              13               7992.20   
1                      124              15               9015.40   
2                      114              19               9062.45   
6                      128              23              10566.70   
15                     158              50              12324.10   
13                     105              37               8876.25   
14                     118              48               9221.40   
12                      73              35               6165.45   
4                       43              20               2749.65   
7                       32              23               2164.70   
8                       12              27               1071.65   
5                        8              21                639.60   
3                        1              20                 50.15   
11                       1              33                 71.00   

    Monthly Revenue After Churn  Percentage difference (%)  
10                     51210.50                      46.25  
9                      53098.85                      42.15  
0                      41949.45                      17.39  
1                      40794.40                      19.90  
2                      32634.85                      24.38  
6                      31125.05                      29.02  
15                     12516.70                      65.98  
13                     13874.90                      48.47  
14                      9054.95                      67.48  
12                     11632.55                      41.90  
4             

#### Financial Analysis Sorted By Monthly Revenue after Churn

In [31]:
Summary_df_sorted = Summary_df.sort_values(by=['Monthly Revenue After Churn'], ascending=False)
Summary_df_sorted

Demographic Segment  Customer Count  Average Tenure (Months)  \
9           Segment 10            1381                    22.89   
10          Segment 11            1338                    22.96   
0            Segment 1             822                    41.57   
1            Segment 2             844                    40.60   
2            Segment 3             596                    43.32   
6            Segment 7             567                    43.98   
13          Segment 14             281                    39.15   
15          Segment 16             317                    24.89   
12          Segment 13             209                    42.33   
14          Segment 15             244                    25.43   
4            Segment 5             211                    25.14   
7            Segment 8             142                    22.04   
8            Segment 9              44                    49.45   
5            Segment 6              39                    44.26   
3            Segment 4               5                    22.60   
11          Segment 12               3                    21.33   

    Average Monthly Charge  Monthly Revenue Before Churn  \
9                    58.98                      81453.15   
10                   61.30                      82016.35   
0                    60.76                      49941.65   
1                    59.02                      49809.80   
2                    69.96                      41697.30   
6                    73.53                      41691.75   
13                   80.96                      22751.15   
15                   78.36                      24840.80   
12                   85.16                      17798.00   
14                   74.90                      18276.35   
4                    49.83                      10513.15   
7                    55.20                       7838.60   
8                    84.80                       3731.30   
5                    80.87                       3153.90   
3                    65.14                        325.70   
11                   92.55                        277.65   

    Churned Customer Count  Churn Rate (%)  Monthly Revenue Lost  \
9                      418              30              28354.30   
10                     429              32              30805.85   
0                      105              13               7992.20   
1                      124              15               9015.40   
2                      114              19               9062.45   
6                      128              23              10566.70   
13                     105              37               8876.25   
15                     158              50              12324.10   
12                      73              35               6165.45   
14                     118              48               9221.40   
4                       43              20               2749.65   
7                       32              23               2164.70   
8                       12              27               1071.65   
5                        8              21                639.60   
3                        1              20                 50.15   
11                       1              33                 71.00   

    Monthly Revenue After Churn  Percentage difference (%)  
9                      53098.85                      42.15  
10                     51210.50                      46.25  
0                      41949.45                      17.39  
1                      40794.40                      19.90  
2                      32634.85                      24.38  
6                      31125.05                      29.02  
13                     13874.90                      48.47  
15                     12516.70                      65.98  
12                     11632.55                      41.90  
14                      9054.95                      67.48  
4             

#### Financial analysis sort by Average Monthly Charge

In [32]:
Summary_df_sorted = Summary_df.sort_values(by=['Average Monthly Charge'], ascending=False)
Summary_df_sorted

Demographic Segment  Customer Count  Average Tenure (Months)  \
11          Segment 12               3                    21.33   
12          Segment 13             209                    42.33   
8            Segment 9              44                    49.45   
13          Segment 14             281                    39.15   
5            Segment 6              39                    44.26   
15          Segment 16             317                    24.89   
14          Segment 15             244                    25.43   
6            Segment 7             567                    43.98   
2            Segment 3             596                    43.32   
3            Segment 4               5                    22.60   
10          Segment 11            1338                    22.96   
0            Segment 1             822                    41.57   
1            Segment 2             844                    40.60   
9           Segment 10            1381                    22.89   
7            Segment 8             142                    22.04   
4            Segment 5             211                    25.14   

    Average Monthly Charge  Monthly Revenue Before Churn  \
11                   92.55                        277.65   
12                   85.16                      17798.00   
8                    84.80                       3731.30   
13                   80.96                      22751.15   
5                    80.87                       3153.90   
15                   78.36                      24840.80   
14                   74.90                      18276.35   
6                    73.53                      41691.75   
2                    69.96                      41697.30   
3                    65.14                        325.70   
10                   61.30                      82016.35   
0                    60.76                      49941.65   
1                    59.02                      49809.80   
9                    58.98                      81453.15   
7                    55.20                       7838.60   
4                    49.83                      10513.15   

    Churned Customer Count  Churn Rate (%)  Monthly Revenue Lost  \
11                       1              33                 71.00   
12                      73              35               6165.45   
8                       12              27               1071.65   
13                     105              37               8876.25   
5                        8              21                639.60   
15                     158              50              12324.10   
14                     118              48               9221.40   
6                      128              23              10566.70   
2                      114              19               9062.45   
3                        1              20                 50.15   
10                     429              32              30805.85   
0                      105              13               7992.20   
1                      124              15               9015.40   
9                      418              30              28354.30   
7                       32              23               2164.70   
4                       43              20               2749.65   

    Monthly Revenue After Churn  Percentage difference (%)  
11                       206.65                      29.32  
12                     11632.55                      41.90  
8                       2659.65                      33.54  
13                     13874.90                      48.47  
5                       2514.30                      22.57  
15                     12516.70                      65.98  
14                      9054.95                      67.48  
6                      31125.05                      29.02  
2                      32634.85                      24.38  
3                        275.55                      16.68  
10            

#### Financial analysis sort by Percentage difference between the Monthly Revenue before churn and after churned

In [33]:
Summary_df_sorted = Summary_df.sort_values(by=['Percentage difference (%)'], ascending=False)
Summary_df_sorted

Demographic Segment  Customer Count  Average Tenure (Months)  \
14          Segment 15             244                    25.43   
15          Segment 16             317                    24.89   
13          Segment 14             281                    39.15   
10          Segment 11            1338                    22.96   
9           Segment 10            1381                    22.89   
12          Segment 13             209                    42.33   
8            Segment 9              44                    49.45   
7            Segment 8             142                    22.04   
4            Segment 5             211                    25.14   
11          Segment 12               3                    21.33   
6            Segment 7             567                    43.98   
2            Segment 3             596                    43.32   
5            Segment 6              39                    44.26   
1            Segment 2             844                    40.60   
0            Segment 1             822                    41.57   
3            Segment 4               5                    22.60   

    Average Monthly Charge  Monthly Revenue Before Churn  \
14                   74.90                      18276.35   
15                   78.36                      24840.80   
13                   80.96                      22751.15   
10                   61.30                      82016.35   
9                    58.98                      81453.15   
12                   85.16                      17798.00   
8                    84.80                       3731.30   
7                    55.20                       7838.60   
4                    49.83                      10513.15   
11                   92.55                        277.65   
6                    73.53                      41691.75   
2                    69.96                      41697.30   
5                    80.87                       3153.90   
1                    59.02                      49809.80   
0                    60.76                      49941.65   
3                    65.14                        325.70   

    Churned Customer Count  Churn Rate (%)  Monthly Revenue Lost  \
14                     118              48               9221.40   
15                     158              50              12324.10   
13                     105              37               8876.25   
10                     429              32              30805.85   
9                      418              30              28354.30   
12                      73              35               6165.45   
8                       12              27               1071.65   
7                       32              23               2164.70   
4                       43              20               2749.65   
11                       1              33                 71.00   
6                      128              23              10566.70   
2                      114              19               9062.45   
5                        8              21                639.60   
1                      124              15               9015.40   
0                      105              13               7992.20   
3                        1              20                 50.15   

    Monthly Revenue After Churn  Percentage difference (%)  
14                      9054.95                      67.48  
15                     12516.70                      65.98  
13                     13874.90                      48.47  
10                     51210.50                      46.25  
9                      53098.85                      42.15  
12                     11632.55                      41.90  
8                       2659.65                      33.54  
7                       5673.90                      32.04  
4                       7763.50                      30.09  
11                       206.65                      29.32  
6             

### Analyzing Segment 11 Services

In [19]:
Segment_11_df = df[unique_demo_df.loc[10, 'Filter']]

In [20]:
Segment_11_service_df = Segment_11_df[['PhoneService', 'MultipleLines', 'InternetService', 'OnlineSecurity', 'OnlineBackup',
                                       'DeviceProtection','TechSupport','StreamingTV', 'StreamingMovies', 'Churn']]

Segment_11_service_df.head()

PhoneService     MultipleLines InternetService       OnlineSecurity  \
4           Yes                No     Fiber optic                   No   
5           Yes               Yes     Fiber optic                   No   
7            No  No phone service             DSL                  Yes   
16          Yes                No              No  No internet service   
19          Yes                No     Fiber optic                   No   

           OnlineBackup     DeviceProtection          TechSupport  \
4                    No                   No                   No   
5                    No                  Yes                   No   
7                    No                   No                   No   
16  No internet service  No internet service  No internet service   
19                  Yes                  Yes                   No   

            StreamingTV      StreamingMovies Churn  
4                    No                   No   Yes  
5                   Yes                  Yes   Yes  
7                    No                   No    No  
16  No internet service  No internet service    No  
19                   No                  Yes    No

In [21]:
Segment_11_service_df.shape

(1338, 10)

In [22]:
#Segment 11 Service comprehensive analysis 
Services = ['PhoneService', 'MultipleLines', 'InternetService', 'OnlineSecurity', 'OnlineBackup', 
           'DeviceProtection','TechSupport','StreamingTV', 'StreamingMovies']

Service_summary = {
    'Service' : [],
    'Customer Count Service User' : [],
    'Service User Churn rate (%)' : [],
    'Customer Count Non Service User' : [],
    'Non_Service User Churn rate (%)' : []
}


for service in Services:
    if service != 'InternetService':
        Service_summary['Service'].append(service)
        dummy_user = Segment_11_service_df[Segment_11_service_df[service]=='Yes']
        
        user_churn_rate = round(((dummy_user[dummy_user['Churn']=='Yes'].shape[0])/(dummy_user.shape[0]))*100, 2)
    
        Service_summary['Customer Count Service User'].append(dummy_user.shape[0])
        Service_summary['Service User Churn rate (%)'].append(user_churn_rate)

        dummy_non_user = Segment_11_service_df[Segment_11_service_df[service]=='No']
        non_user_churn_rate = round(((dummy_non_user[dummy_non_user['Churn']=='Yes'].shape[0])/(dummy_non_user.shape[0]))*100, 2)

        Service_summary['Customer Count Non Service User'].append(dummy_non_user.shape[0])
        Service_summary['Non_Service User Churn rate (%)'].append(non_user_churn_rate)

    else:
        Service_summary['Service'].append(service)
        dummy_user = Segment_11_service_df[(Segment_11_service_df[service]=='Fiber optic')|(Segment_11_service_df[service]=='DSL')]
        
        user_churn_rate = round(((dummy_user[dummy_user['Churn']=='Yes'].shape[0])/(dummy_user.shape[0]))*100, 2)
    
        Service_summary['Customer Count Service User'].append(dummy_user.shape[0])
        Service_summary['Service User Churn rate (%)'].append(user_churn_rate)

        dummy_non_user = Segment_11_service_df[(Segment_11_service_df[service]=='No')]
        non_user_churn_rate = round(((dummy_non_user[dummy_non_user['Churn']=='Yes'].shape[0])/(dummy_non_user.shape[0]))*100, 2)

        Service_summary['Customer Count Non Service User'].append(dummy_non_user.shape[0])
        Service_summary['Non_Service User Churn rate (%)'].append(non_user_churn_rate)

Service_summary_df = pd.DataFrame(Service_summary)
Service_summary_df_sorted = Service_summary_df.sort_values(by=['Customer Count Service User'], ignore_index = True, ascending = False)
Service_summary_df_sorted

Service  Customer Count Service User  Service User Churn rate (%)  \
0      PhoneService                         1203                        32.92   
1   InternetService                         1042                        38.20   
2     MultipleLines                          455                        34.95   
3   StreamingMovies                          444                        37.16   
4       StreamingTV                          425                        38.12   
5      OnlineBackup                          385                        27.27   
6  DeviceProtection                          381                        25.72   
7       TechSupport                          333                        21.02   
8    OnlineSecurity                          326                        21.78   

   Customer Count Non Service User  Non_Service User Churn rate (%)  
0                              135                            24.44  
1                              296                            10.47  
2                              748                            31.68  
3                              598                            38.96  
4                              617                            38.25  
5                              657                            44.60  
6                              661                            45.39  
7                              709                            46.26  
8                              716                            45.67

### Services Analysis for each unique demographic

In [23]:
def analyze_services(df, Unique_demo_df):
    """
    Analyzes the services used by each demographic segment.

    Args:
        df (pd.DataFrame): The main customer data DataFrame with service columns.
        unique_demo_df (pd.DataFrame): DataFrame defining the demographic segments
                                        with 'Group' and 'Filter' columns.

    Returns:
        pd.DataFrame: A DataFrame showing the service usage breakdown for each segment.
    """
    service_columns = [
        'PhoneService', 'MultipleLines', 'InternetService', 'OnlineSecurity',
        'OnlineBackup', 'DeviceProtection', 'TechSupport', 'StreamingTV',
        'StreamingMovies'
    ]
    segment_service_analysis = []

    for i in range(len(unique_demo_df)):
        group_name = unique_demo_df.loc[i, 'Demographic Segment']
        filter_condition = unique_demo_df.loc[i, 'Filter']
        segment_df = df[filter_condition].copy()  # Use a copy to avoid SettingWithCopyWarning

        segment_summary = {'Demographic Segment': group_name, 'Customer Count': segment_df.shape[0],
                           'Customer Percentage' : f'{round((segment_df.shape[0]/df.shape[0])*100, 2)}%',
                           'Churn rate': f'{round((segment_df[segment_df["Churn"] == "Yes"].shape[0]/segment_df.shape[0])*100,2)}%' }
                                                   
        for service in service_columns:
            if service in segment_df.columns:
                # For binary services (Yes/No, or similar)
                if segment_df[service].nunique() <= 2 and 'Yes' in segment_df[service].unique():
                    #information on the services user
                    dummy_user = segment_df[segment_df[service] == 'Yes']
                    yes_count = dummy_user.shape[0]
                    churn_count = dummy_user[dummy_user['Churn'] == 'Yes'].shape[0]

                    #information on the non services users churning
                    dummy_non_user = segment_df[segment_df[service] == 'No']
                    non_user_count = dummy_non_user.shape[0]
                    non_churn_count = dummy_non_user[dummy_non_user['Churn'] == 'Yes'].shape[0]
                    
                    segment_summary[f'Has {service}'] = f'{yes_count} ({round((yes_count / segment_df.shape[0]) * 100, 2)}%)'
                    segment_summary[f'{service} Churn rate'] = f'{round((churn_count/yes_count)*100, 2)}%'

                    if non_churn_count != 0: 
                        segment_summary[f'{service} Non_User Churn rate'] = f'{round((non_churn_count/non_user_count)*100, 2)}%'
                    else:
                        segment_summary[f'{service} Non_User Churn rate'] = 'N/A'
                
                # For services with multiple options (e.g., InternetService: DSL, Fiber optic, No)
                elif service == 'InternetService':
                    service_counts = segment_df[service].value_counts().to_dict()
                    if len(service_counts) == 3:
                        for option, count in service_counts.items():
                            dummy_user = segment_df[segment_df[service] == option]
                            yes_count = dummy_user.shape[0]
                            churn_count = dummy_user[dummy_user['Churn']=='Yes'].shape[0]
                            
                            segment_summary[f'Internet - {option}'] = f'{yes_count} ({round((yes_count / segment_df.shape[0]) * 100, 2)}%)'
                            
                            if churn_count != 0:
                                segment_summary[f'Internet - {option} Churn rate'] = f'{round((churn_count/yes_count)*100, 2)}%'
                            else:
                                segment_summary[f'Internet - {option} Churn rate'] = 'N/A'
                            
                    else:
                        options = ['DSL', 'Fiber optic', 'No']
                        for option in options:
                            if option in service_counts.keys():
                                pass
                            else:
                                service_counts.update({option: 0})
                        for option, count in service_counts.items():
                            dummy_user = segment_df[segment_df[service]== option]
                            yes_count = dummy_user.shape[0]
                            churn_count = dummy_user[dummy_user['Churn']=='Yes'].shape[0]

                            if yes_count != 0:
                                segment_summary[f'Internet - {option}'] = f'{yes_count} ({round((yes_count / segment_df.shape[0]) * 100, 2)}%)'
                            else:
                                segment_summary[f'Internet - {option}'] = 'N/A'
                            
                            if churn_count != 0:
                                segment_summary[f'Internet - {option} Churn rate'] = f'{round((churn_count/yes_count)*100, 2)}%'
                            else:
                                segment_summary[f'Internet - {option} Churn rate'] = 'N/A'
                        
                # Add other multi-option service handling if needed
                else:
                    #segment_summary[f'Usage of {service}'] = segment_df[service].mode().iloc[0] if not segment_df[service].empty else 'No Usage'
                    dummy_user = segment_df[segment_df[service] == 'Yes']
                    yes_count = dummy_user.shape[0]
                    churn_count = dummy_user[dummy_user['Churn'] == 'Yes'].shape[0]

                    #information on the non services users churning
                    dummy_non_user = segment_df[segment_df[service] == 'No']
                    non_user_count = dummy_non_user.shape[0]
                    non_churn_count = dummy_non_user[dummy_non_user['Churn'] == 'Yes'].shape[0]
                    
                    segment_summary[f'Has {service}'] = f'{yes_count} ({round((yes_count / segment_df.shape[0]) * 100, 2)}%)'
                    
                    if churn_count != 0:
                        segment_summary[f'{service} Churn rate'] = f'{round((churn_count/yes_count)*100, 2)}%'
                    else:
                        segment_summary[f'{service} Churn rate'] = 'N/A'

                    if non_churn_count != 0:
                        segment_summary[f'{service} Non_User Churn rate'] = f'{round((non_churn_count/non_user_count)*100, 2)}%'
                    else:
                        segment_summary[f'{service} Non_User Churn rate'] = 'N/A'
                        
            else:
                segment_summary[f'Has {service}'] = 'N/A'
            

        segment_service_analysis.append(segment_summary)

    return pd.DataFrame(segment_service_analysis)

service_usage_analysis = analyze_services(df, Unique_demo)
service_usage_analysis.to_csv('Telco-Customer-Segment-Analysis.csv', index = False)
service_usage_analysis

Demographic Segment  Customer Count Customer Percentage Churn rate  \
0            Segment 1             822              11.67%     12.77%   
1            Segment 2             844              11.98%     14.69%   
2            Segment 3             596               8.46%     19.13%   
3            Segment 4               5               0.07%      20.0%   
4            Segment 5             211                3.0%     20.38%   
5            Segment 6              39               0.55%     20.51%   
6            Segment 7             567               8.05%     22.57%   
7            Segment 8             142               2.02%     22.54%   
8            Segment 9              44               0.62%     27.27%   
9           Segment 10            1381              19.61%     30.27%   
10          Segment 11            1338               19.0%     32.06%   
11          Segment 12               3               0.04%     33.33%   
12          Segment 13             209               2.97%     34.93%   
13          Segment 14             281               3.99%     37.37%   
14          Segment 15             244               3.46%     48.36%   
15          Segment 16             317                4.5%     49.84%   

   Has PhoneService PhoneService Churn rate PhoneService Non_User Churn rate  \
0      749 (91.12%)                  13.08%                            9.59%   
1      764 (90.52%)                  15.05%                           11.25%   
2      544 (91.28%)                   18.2%                           28.85%   
3        5 (100.0%)                   20.0%                              N/A   
4      186 (88.15%)                  20.97%                            16.0%   
5       33 (84.62%)                  24.24%                              N/A   
6       512 (90.3%)                  22.27%                           25.45%   
7      123 (86.62%)                  24.39%                           10.53%   
8       41 (93.18%)                  26.83%                           33.33%   
9     1242 (89.93%)                  30.27%                           30.22%   
10    1203 (89.91%)                  32.92%                           24.44%   
11       3 (100.0%)                  33.33%                              N/A   
12     198 (94.74%)                  34.34%                           45.45%   
13     250 (88.97%)                   38.0%                           32.26%   
14     219 (89.75%)                   48.4%                            48.0%   
15     289 (91.17%)                  49.13%                           57.14%   

   Has MultipleLines MultipleLines Churn rate  \
0       342 (41.61%)                   13.74%   
1       353 (41.82%)                   15.01%   
2       310 (52.01%)                    20.0%   
3          1 (20.0%)                     0.0%   
4        60 (28.44%)                   23.33%   
5        22 (56.41%)                   13.64%   
6       308 (54.32%)                   23.38%   
7        40 (28.17%)                    20.0%   
8        31 (70.45%)                   29.03%   
9       438 (31.72%)                   35.39%   
10      455 (34.01%)                   34.95%   
11        2 (66.67%)                     0.0%   
12      140 (66.99%)                   35.71%   
13      176 (62.63%)                   40.91%   
14      129 (52.87%)                   50.39%   
15      164 (51.74%)                   49.39%   

   MultipleLines Non_User Churn rate  ...  \
0                             12.53%  ...   
1                             15.09%  ...   
2                             15.81%  ...   
3                              25.0%  ...   
4                             19.84%  ...   
5                             45.45%  ...   
6                             20.59%  ...   
7                             26.51%  ...   
8                              20.0%  ...   
9                             27.49%  ...   
10                            31.68%  ...   
11                        

## Create a new data-set assigning each customer to segment based on the demographic 

In [24]:
def assign_segments(df, unique_demo_df):
    """
    Assigns a 'Segment' label to each customer in the df
    based on the demographic criteria in unique_demo_df.

    Args:
        df (pd.DataFrame): The main customer data DataFrame.
        unique_demo_df (pd.DataFrame): DataFrame defining the demographic segments
                                        with 'Demographic Segment' and 'Filter' columns.

    Returns:
        pd.DataFrame: The original DataFrame with an added 'Segment' column.
    """
    df.insert(5, 'Segment' , None)  # Initialize the 'Segment' column with None

    for index, row in unique_demo_df.iterrows():
        segment_name = row['Demographic Segment']
        filter_condition = row['Filter']  # Ensure it's treated as a string
        
        # Apply the filter condition using df.query() with the string
        try:
            segment_customers = df[filter_condition].index
            # Assign the segment name to these customers in the 'Segment' column
            df.loc[segment_customers, 'Segment'] = segment_name
        except Exception as e:
            print(f"Error applying filter ")

    return df

# Assuming you have your 'df' and 'unique_demo_df' DataFrames loaded
# Example of your original df (replace with your actual DataFrame)


df_with_segment = assign_segments(df.copy(), unique_demo_df.copy()) # Use .copy()
df_with_segment.to_csv('Telco-Customer-Churn-Data-Update.csv', index = False)
df_with_segment.head()

customerID  gender  SeniorCitizen Partner Dependents     Segment  tenure  \
0  7590-VHVEG  Female              0     Yes         No   Segment 3       1   
1  5575-GNVDE    Male              0      No         No  Segment 10      34   
2  3668-QPYBK    Male              0      No         No  Segment 10       2   
3  7795-CFOCW    Male              0      No         No  Segment 10      45   
4  9237-HQITU  Female              0      No         No  Segment 11       2   

  PhoneService     MultipleLines InternetService  ... DeviceProtection  \
0           No  No phone service             DSL  ...               No   
1          Yes                No             DSL  ...              Yes   
2          Yes                No             DSL  ...               No   
3           No  No phone service             DSL  ...              Yes   
4          Yes                No     Fiber optic  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   
4          No          No              No  Month-to-month              Yes   

               PaymentMethod MonthlyCharges TotalCharges  Churn  
0           Electronic check          29.85        29.85     No  
1               Mailed check          56.95      1889.50     No  
2               Mailed check          53.85       108.15    Yes  
3  Bank transfer (automatic)          42.30      1840.75     No  
4           Electronic check          70.70       151.65    Yes  

[5 rows x 22 columns]

## Findings

### Customer Demographic Analysis & Churn Impact on Revenue

### **1. Introduction**
This report examines the impact of different customer demographics on company revenue and service consumption, both before and after customer churn. The analysis is based on **16 unique demographic segments**, categorized by attributes such as gender, senior citizen status, partner, and dependents.

### **2. Revenue Analysis**

#### **Top 3 Segments by Monthly Revenue (Pre-Churn)**
| Segment | Description | Monthly Revenue (£) |
|---------|---------------------------|----------------|
| **Segment 11** | Young, single males without dependents | **82,016.35** |
| **Segment 10** | Young, single females without dependents | **81,453.15** |
| **Segment 1**  | Young, partnered males with dependents | **49,941.65** |

#### **Top 3 Segments by Revenue Lost Due to Churn**
| Segment | Description | Revenue Lost (£) |
|---------|---------------------------|----------------|
| **Segment 11** | Young, single males without dependents | **30,805.85** |
| **Segment 10** | Young, single females without dependents | **28,354.30** |
| **Segment 16** | Old, single females without dependents | **12,324.10** |

#### **Top 3 Segments by Monthly Revenue Post-Churn**
| Segment | Description | Monthly Revenue (£) |
|---------|---------------------------|----------------|
| **Segment 10** | Young, single females without dependents | **53,998.85** |
| **Segment 11** | Young, single males without dependents | **51,210.50** |
| **Segment 1**  | Young, partnered males with dependents | **41,949.45** |

#### **Top 3 Segments by Average Monthly Charges**
| Segment | Description | Average Monthly Charge (£) |
|---------|---------------------------|----------------|
| **Segment 12** | Old, single females with dependents | **92.55** |
| **Segment 13** | Old, partnered females without dependents | **85.16** |
| **Segment 9**  | Old, partnered males with dependents | **84.80** |

#### **Top 3 Segments by Percentage Difference in Revenue Pre- & Post-Churn**
| Segment | Description | Revenue Loss (%) | Churn Rate (%) |
|---------|---------------------------|---------------|--------------|
| **Segment 15** | Old, single males without dependents | **67.48%** | **48%** |
| **Segment 16** | Old, single females without dependents | **65.98%** | **50%** |
| **Segment 14** | Old, partnered males with dependents | **48.47%** | **37%** |

### **3. Key Segments for Business Focus**
The following segments significantly impact financial performance and churn rates:
- **Segment 1:** Young, partnered males with dependents
- **Segment 10:** Young, single females without dependents
- **Segment 11:** Young, single males without dependents
- **Segment 16:** Old, single females without dependents

### **4. Service Usage & Churn Rate Analysis**

#### **Segment 1 – Young, Partnered Males with Dependents**
- **Phone services:** 91.12% usage (churn rate **13.08%**).
- **Internet services:**
  - DSL: **38.08% usage**, churn rate **10.22%**.
  - Fibre: **31.87% usage**, churn rate **25.95%**.
  - No internet: churn rate **2.02%** (significantly lower).
- **Tech support:** **34.79% usage**, churn rate **7.69%** (vs. **26.99%** without).
- **Online security, backup & device protection:** All reduce churn rates.
- **Streaming services (TV & Movies):** Higher churn rates compared to non-users.

#### **Segment 10 – Young, Single Females without Dependents**
- **Phone services:** 89.93% usage, churn rate **30.27%**.
- **Internet services:**
  - Fibre users have **44.71% churn**, significantly higher than DSL users (**27.12%**).
  - No internet: churn rate **12.0%** (lowest).
- **Tech support & security services:** Lower churn among users.
- **Streaming services:** Both **TV & Movies** have churn rates of **~36%**.

#### **Segment 11 – Young, Single Males without Dependents**
- **Phone services:** 89.91% usage, churn rate **32.92%**.
- **Internet services:**
  - Fibre users: **51.44% churn** (highest among all segments).
  - No internet: churn rate **10.47%** (lowest).
- **Tech support & security services:** Lower churn among users.
- **Streaming services:** Both **TV & Movies** have **~37-38% churn rates**.

#### **Segment 16 – Old, Single Females without Dependents**
- **Phone services:** 91.17% usage, churn rate **49.13%**.
- **Internet services:**
  - Fibre users: **54.66% churn** (highest).
  - No internet: churn rate **20%** (lowest).
- **Tech support & security services:** Churn rates **significantly lower** among users.
- **Streaming services:** **45-48% churn**, showing minimal retention benefits.

### **5. Business Insights & Recommendations**
1. **Target High-Churn Segments with Retention Strategies:**
   - **Segment 10 & 11** have high churn but also generate substantial revenue.
   - **Segment 16 has extreme churn rates**—consider incentives for older customers.
   - **Internet services (especially Fibre users) are at risk**—investigate pain points.

2. **Enhance Service Bundling & Value Addition:**
   - **Tech support significantly reduces churn**—increase awareness or promotions.
   - **Security services (Online Backup, Device Protection, etc.) lower churn**—consider bundled offers.

3. **Revenue Optimization Strategies:**
   - High-churn **segments should receive personalized retention efforts**—discounts, loyalty programs, or improved customer experience.
   - Focus on **Segment 1 for stability**, as it retains strong revenue while maintaining lower churn.

### **6. Conclusion**
This analysis highlights critical revenue segments and their churn patterns, identifying key services that impact customer retention. **Retention strategies should focus on high-churn but high-revenue segments, especially Fibre internet users and customers without security services.**